src/common/app-paths/app-path-injection-token.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import { getInjectionToken } from "@ogre-tools/injectable";
+import type { PathName } from "./app-path-names";
+import { createChannel } from "../ipc-channel/create-channel/create-channel";
+
+export type AppPaths = Record<PathName, string>;
+
+export const appPathsInjectionToken = getInjectionToken<AppPaths>();
```

Comment:
What is injection token and where/how it can be used?

Reply:
It solves the problem:

- Common code wants to inject interface of something that is environment specific
- Common code shouldn't know anything about environment specifics (e.g. `if (ipcMain) {..`) 

In this case  common code wants to inject `appPaths` which have two separate implementations:

- In `main` we get app paths from `electron`
- In `renderer` we get app paths from `main`

So in a nutshell:

- Injection Token is something that can be injected, but it doesn't have implementation. It defines the interface that the implementation needs to return
- Any other `injectable` can choose to implement `injectionToken` by adding `injectionToken: someInjectionToken` property.